xbmc/CompileInfo.cpp.in

```diff
@@ -55,6 +55,11 @@ const char* CCompileInfo::GetSCMID()
   return "@APP_SCMID@";
 }
 
+const std::string CCompileInfo::GetTrustedAddonRepos()
```

Comment:
Either return `const std::string&` or `std::string` (without const); never return plain `const`

Reply:
ok, happy to go with `const std::string&` - could you please explain why never return plain `const`? ( only if two sentences are enough, otherwise i'll have to read ;) )

Reply:
for built-in types (like `bool`), it has no effect and is thus confusing to the programmer

for custom types (like `std::string`), it unnecessarily restricts the things the user of the function can do with it (e.g., the return value cannot be used as lvalue, and the user could not append to the string although it would be a copy anyway)

Reply:
alright, thank you